x-pack/legacy/plugins/infra/public/containers/logs/log_analysis/log_analysis_setup_state.tsx

```diff
@@ -60,24 +96,48 @@ export const useAnalysisSetupState = ({
     return cleanupAndSetupModule(selectedIndexNames, startTime, endTime);
   }, [cleanupAndSetupModule, selectedIndexNames, startTime, endTime]);
 
-  const validationErrors: ValidationErrors[] = useMemo(
+  const isValidating = useMemo(
     () =>
-      Object.values(selectedIndices).some(isSelected => isSelected)
-        ? []
-        : ['TOO_FEW_SELECTED_INDICES' as const],
-    [selectedIndices]
+      validateIndicesRequest.state === 'pending' ||
+      validateIndicesRequest.state === 'uninitialized',
+    [validateIndicesRequest.state]
   );
 
+  const validationErrors = useMemo<ValidationIndicesUIError[]>(() => {
+    if (isValidating) {
+      return [];
+    }
+
+    if (validateIndicesRequest.state === 'rejected') {
+      return [{ error: 'NETWORK_ERROR' }];
+    }
+
+    if (selectedIndexNames.length === 0) {
+      return [{ error: 'TOO_FEW_SELECTED_INDICES' }];
+    }
+
+    const indicesWithErrors = validatedIndices.filter(
+      index => selectedIndexNames.includes(index.index) && index.validation !== undefined
+    );
+
+    if (indicesWithErrors.length > 0) {
+      return indicesWithErrors.map(index => index.validation!);
```

Comment:
The `!` is safe here. A couple of lines above I make sure all elements of `indicesWithErrors` have indeed a `.validation`.

I can try to hint the compiler that `indicesWithErrors` only contains elements that have indeed a validation with a different interface so we can skip the `!`.

I prefer to keep the filter and the map in two different operations. I find it easier to reason about.

Reply:
Yes, it might be safe now but the compiler won't help us notice if it becomes unsafe at some point. So it would be great if we could make that clear to the compiler. :+1: